src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -1718,6 +1755,81 @@ private static async Task ExecuteGraphQLIntrospectionQueries(TestServer server,
             }
         }
 
+        private static JsonContent GetJsonContentForCosmosConfigRequest(string endpoint, string config = null, bool useAccessToken = false)
+        {
+            if (CONFIGURATION_ENDPOINT == endpoint)
+            {
+                ConfigurationPostParameters configParams = GetCosmosConfigurationParameters();
+                if (config != null)
+                {
+                    configParams = configParams with { Configuration = config };
+                }
+
+                if (useAccessToken)
+                {
+                    configParams = configParams with
+                    {
+                        ConnectionString = "AccountEndpoint=https://localhost:8081/;",
+                        AccessToken = GenerateMockJwtToken()
+                    };
+                }
+
+                return JsonContent.Create(configParams);
+            }
+            else if (CONFIGURATION_ENDPOINT_V2 == endpoint)
+            {
+                ConfigurationPostParametersV2 configParams = GetCosmosConfigurationParametersV2();
+                if (config != null)
+                {
+                    configParams = configParams with { Configuration = config };
+                }
+
+                if (useAccessToken)
+                {
+                    // With an invalid access token, when a new instance of CosmosClient is created with that token, it
+                    // won't throw an exception.  But when a graphql request is coming in, that's when it throws a 401
+                    // exception. To prevent this, CosmosClientProvider parses the token and retrieves the "exp" property
+                    // from the token, if it's not valid, then we will throw an exception from our code before it
+                    // initiating a client. Uses a valid fake JWT access token for testing purposes.
+                    RuntimeConfig overrides = new(null, new DataSource(DatabaseType.cosmosdb_nosql), null, null)
+                    {
+                        ConnectionString = "AccountEndpoint=https://localhost:8081/;"
+                    };
+
+                    configParams = configParams with
+                    {
+                        ConfigurationOverrides = JsonSerializer.Serialize(overrides),
+                        AccessToken = GenerateMockJwtToken()
+                    };
+                }
+
+                return JsonContent.Create(configParams);
+            }
+            else
+            {
+                throw new ArgumentException($"Unexpected configuration endpoint. {endpoint}");
+            }
+        }
+
+        private static string GenerateMockJwtToken()
```

Comment:
JwtTokenAuthenticationUnitTests.cs has a similar CreateJwt() function. Could you reuse that here?